llvm/lib/Transforms/IPO/DynUntrustedAllocPre.cpp

```diff
@@ -0,0 +1,273 @@
+//===-- UntrustedAlloc.cpp - UntrustedAlloc Infrastructure ---------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the common initialization infrastructure for the
+// DynUntrustedAlloc library.
+//
+//===----------------------------------------------------------------------===//
+
+#include "llvm/Transforms/IPO/DynUntrustedAllocPre.h"
+#include "llvm/ADT/DenseMap.h"
+#include "llvm/ADT/PostOrderIterator.h"
+#include "llvm/ADT/SCCIterator.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/Analysis/CallGraph.h"
+#include "llvm/Analysis/LazyCallGraph.h"
+#include "llvm/IR/Attributes.h"
+#include "llvm/IR/BasicBlock.h"
+#include "llvm/IR/CFG.h"
+#include "llvm/IR/CallSite.h"
+#include "llvm/IR/DebugInfo.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/InstIterator.h"
+#include "llvm/IR/Instruction.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IR/Type.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include <set>
+#include <string>
+
+#define DEBUG_TYPE "dyn-untrusted"
+
+using namespace llvm;
+
+namespace {
+class IDGenerator {
```

Comment:
Doesn't this class exist elsewhere? why redefine it? it should either be moved into a common library, or made a header only dependency. 

If its different, then it should have at least a different name

Reply:
Reduced IDGenerator definition to only Post as Pre only allocates a placeholder value anyway.